pkg/driver/agent.go

```diff
@@ -379,8 +379,15 @@ func (ns *node) NodeExpandVolume(
 	}
 
 	if err = zfs.ResizeZFSVolume(vol, req.GetVolumePath(), resizefs); err != nil {
+		code := codes.Internal
+
+		// error defined in zfs source code: https://github.com/openzfs/zfs/blob/zfs-2.3.99/lib/libzfs/libzfs_util.c#L551
+		if strings.Contains(err.Error(), "size is greater than available space") {
+			code = codes.ResourceExhausted
```

Comment:
I wonder why the actual resize isn't confirmed in the controller resize, CC @Abhinandan-Purkait

Reply:
Isn't it because the NodeExpand happens later and the job of controller is just to update the CR and the node plugin does the job based on the CR change?

Reply:
Yeah

Reply:
I'd suggest we should also change that, otherwise the offline resize is not really resizing, it's just setting the request, which may fail only when a pod is scheduled to run.

Reply:
I've already opened an issue about it #657